cfgov/research_reports/models.py

```diff
@@ -0,0 +1,235 @@
+            'the fields in the "Report Content" tab.'
+            '<ul class="help">'
+            '    <li>&bull; If you uploaded a new report file for '
+            'processing, check this box.</li>'
+            '    <li>&bull; If you manually edited fields in the '
+            '"Report Content" tab, do not check this box</li>'
+            '</ul>'
+        )
+    )
+
+    # Report Upload Tabs
+    upload_panels = [
+        MultiFieldPanel([
+            FieldPanel('title'),
+        ], heading="Report Title"),
+        MultiFieldPanel([
+            DocumentChooserPanel('report_file'),
+            FieldPanel('process_report'),
+        ], heading='Report Document'),
+    ]
+
+    # Report Content Fields
+    base_form_class = ReportForm
+    report_type = models.CharField(max_length=100, blank=True)
+    header = models.CharField(max_length=200, blank=True)
+    subheader = models.TextField(blank=True)
+    pdf_location = models.CharField(max_length=150, blank=True)
+    footnotes = models.TextField(blank=True)
+
+    # Report content tab
+    content_panels = [
+        MultiFieldPanel([
+            FieldPanel('report_type'),
+            FieldPanel('header'),
+            FieldPanel('subheader'),
+            InlinePanel(
+                'report_authors',
+                label='Author Name',
+                min_num=0
+            ),
+            FieldPanel('pdf_location')
+        ], heading='Report Header'),
+        InlinePanel(
+            'report_sections',
+            label='Section',
+            min_num=0
+        ),
+        FieldPanel('footnotes')
+    ]
+
+    sidefoot_panels = CFGOVPage.sidefoot_panels
+
+    # Tab handler interface
+    edit_handler = TabbedInterface([
+        ObjectList(upload_panels, heading='Report Upload'),
+        ObjectList(content_panels, heading='Report Content'),
+        ObjectList(sidefoot_panels, heading='Sidebar'),
+        ObjectList(CFGOVPage.settings_panels, heading='Configuration'),
+    ])
+
+    template = 'index.html'
+
+    objects = PageManager()
+
+    def parse_report_file(self):
+        if self.report_file and self.process_report:
+            deploy_env = _get_deploy_environment()
+            if os.environ.get("JENKINS_API_ENABLED"):
+                management.call_command('trigger_jenkins_build', self.id)
+            elif deploy_env == 'local':
+                management.call_command('parse_research_report', self.id)
+            else:
+                # deployed to another environment where s3 isn't configured
+                # TODO: show a friendly error message to the user
+                pass
+
+    def get_context(self, request, *args, **kwargs):
+        context = super(ResearchReportPage, self) \
+            .get_context(request, *args, **kwargs)
+        context.update({
+            'get_report_sections': get_report_sections,
+            'get_report_appendices': get_report_appendices,
+            'get_researchers': get_researchers
```

Comment:
These methods get called in the template to generate variables; would it be simpler to just call those here and pass the variables in directly?